### ISSUE TITLE
[css-2026] Add specs to Reliable Candidate Recommendations or Official Definition

### ISSUE THREAD

**Issue by SebastianZ** [w3c/csswg-drafts#9770]
There are a bunch of specifications which might be promoted to be in the Official Definition section.

# [Media Queries 4](https://www.w3.org/TR/mediaqueries-4/)

Current status: CRD (last published: 25 December 2021)
[WPT](https://wpt.fyi/results/css/mediaqueries?label=stable&label=master&aligned): passing tests: Chrome: 1671, Edge: 1671, Firefox: 1671, Safari: 1668; [interop](https://wpt.fyi/results/css/mediaqueries?label=master&label=stable&aligned&q=all(status%3Apass)): 1659 / 1695 (97.9%)
Current Snapshot section: Fairly Stable
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Amediaqueries-4): 21, [in spec.](https://drafts.csswg.org/mediaqueries-4/#issues-index): 0

Notes: Very high interoperability, spec. in CRD, though many open issues. So it might also be put into Rough Interoperability, instead.

# [CSS Scroll Snap 1](https://www.w3.org/TR/css-scroll-snap-1/)

Current status: CR (last published: 11 March 2021)
[WPT](https://wpt.fyi/results/css/css-scroll-snap?label=master&label=stable&aligned): passing tests (stable): Chrome: 693, Edge: 685, Firefox: 640, Safari: 637; [interop](https://wpt.fyi/results/css/css-scroll-snap?label=master&label=stable&aligned&q=all(status%3Apass)): 632 / 698 (90.5%)
Current Snapshot section: Fairly Stable
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-scroll-snap-1): 20, [in spec.](https://drafts.csswg.org/css-scroll-snap-1/#issues-index): 0

Notes: High interooperability, spec. in CR, though many open issues. So it might also be put to Rough Interoperability, instead.

# [CSS Scrollbars 1](https://www.w3.org/TR/css-scrollbars-1/)

Current status: CR (with [request to move it to PR](https://github.com/w3c/csswg-drafts/issues/9508); last published: 9 December 2021)
[WPT](https://wpt.fyi/results/css/css-scrollbars?label=master&label=experimental&aligned): passing tests (experimental): Chrome: 113, Edge: 113, Firefox: 113, Safari: 51; [interop](https://wpt.fyi/results/css/css-scrollbars?label=master&label=experimental&aligned&q=all(status%3Apass)): 51 / 113 (45.1%), 113 / 113 (100%) for [Chrome, Edge and Firefox](https://wpt.fyi/results/css/css-scrollbars?label=master&label=experimental&aligned&q=chrome%3Apass%20and%20edge%3Apass%20and%20firefox%3Apass)
Current Snapshot section: Fairly Stable
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-scrollbars-1): 6, [in spec.](https://drafts.csswg.org/css-scrollbars-1/#issues-index): 0

Notes: 100% interoperability between two engines, spec. in CR, few open issues.

# [CSS Grid 2](https://www.w3.org/TR/css-grid-2/)

Current status: CRD (last published: 18 December 2020)
[WPT](https://wpt.fyi/results/css/css-grid?label=master&label=stable&aligned&q=none(path%3A%2Fcss%2Fcss-grid%2Fmasonry%2F)): passing tests (stable): Chrome: 12010, Edge: 12010, Firefox: 11268, Safari: 11640; [interop](https://wpt.fyi/results/css/css-grid?label=master&label=stable&aligned&q=none(path%3A%2Fcss%2Fcss-grid%2Fmasonry%2F)%20and%20all(status%3Apass)): 11064 / 12094 (91.5%)
Current Snapshot section: Rough Interoperability
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-grid-2): 12, [in spec.](https://drafts.csswg.org/css-grid-2/#issues-index): 2

Notes: High interoperability, spec. in CRD, still some open issues, last published version outdated. It might also move to Rough Interoperability, instead.
CSS Grid 1 is currently listed under Rough Interoperability, as well, though as it is superseded by Level 2, it can probably be removed.

# [CSS Will Change 1](https://www.w3.org/TR/css-will-change-1/)

Current status: CRD (last published: 5 May 2022)
[WPT](https://wpt.fyi/results/css/css-will-change?label=master&label=stable&aligned): passing tests (stable): Chrome: 162, Edge: 159, Firefox: 170, Safari: 160; [interop](https://wpt.fyi/results/css/css-will-change?label=stable&label=master&aligned&q=all(status%3Apass)): 152 / 170 (89,4%), 162 / 170 (95.3%) for [Chrome and Firefox](https://wpt.fyi/results/css/css-will-change?label=master&label=stable&aligned&q=chrome%3Apass%20and%20firefox%3Apass)
Current Snapshot section: Rough Interoperability
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-will-change-1): 8, [in spec.](https://drafts.csswg.org/css-will-change-1/#issues-index): 0

Notes: Very high interoperability, spec. in CRD, some open issues.

# [Color Adjust 1](https://www.w3.org/TR/css-color-adjust-1/)

Current status: CRD (last published: 14 June 2022)
[WPT](https://wpt.fyi/results/css/css-color-adjust?label=master&label=stable&aligned): passing tests (stable): Chrome: 152, Edge: 152, Firefox: 144, Safari: 89; [interop](https://wpt.fyi/results/css/css-color-adjust?label=stable&label=master&aligned&q=all(status%3Apass)): 75 / 161 (46.6%), 135 / 161 (83.9%) for [Chrome, Edge and Firefox](https://wpt.fyi/results/css/css-color-adjust?label=master&label=stable&aligned&q=chrome%3Apass%20and%20edge%3Apass%20and%20firefox%3Apass)
Current Snapshot section: Rough Interoperability
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-color-adjust-1): 8, [in spec.](https://drafts.csswg.org/css-color-adjust-1/#issues-index): 1

Notes: High interoperability between two engines, spec. in CRD, some open issues.

# [CSS Conditional 4](https://www.w3.org/TR/css-conditional-4/)

Current status: CR (last published: 17 February 2022)
[WPT](https://wpt.fyi/results/css/css-conditional?label=master&label=stable&aligned): passing tests (stable): Chrome: 2768, Edge: 2752, Firefox: 2627, Safari: 2645; [interop](https://wpt.fyi/results/css/css-conditional?label=stable&label=master&aligned&q=all(status%3Apass)): 2573 / 2809 (91.6%)
Current Snapshot section: Rough Interoperability
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-conditional-4): 3, [in spec.](https://drafts.csswg.org/css-conditional-4/#issues-index): 1

Notes: High interoperability, spec. in CR, few open issues.
This spec. is currently a delta spec. against Level 3. And Level 3 is currently part of the Official Definition. So when the rest of Level 3 gets incorporated into Level 4, so it supersedes Level 3, it should replace Level 3 in the Official Definition.

# [Geometry 1](https://www.w3.org/TR/geometry-1/)

Current status: CR (last published: 4 December 2018)
[WPT](https://wpt.fyi/results/css/geometry?label=stable&label=master&aligned): passing tests: Chrome: 1547, Edge: 1551, Firefox: 1539, Safari: 1530; [interop](https://wpt.fyi/results/css/geometry?label=master&label=stable&aligned&q=all(status%3Apass)): 1522 / 1554 (97.9%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/fxtf-drafts/labels/geometry-1): 23, [in spec.](https://drafts.fxtf.org/geometry-1/#issues-index): 0

Notes: Very high interoperability, spec. in CR, though many open issues and last published version outdated. So it might also be put to Rough Interoperability, instead.

# [CSS Shapes 1](https://www.w3.org/TR/css-shapes-1/)

Current status: WD (last published: 15 November 2022)
[WPT](https://wpt.fyi/results/css/css-shapes?label=master&label=stable&aligned): passing tests: Chrome: 5094, Edge: 5012, Firefox: 5789, Safari: 5069; [interop](https://wpt.fyi/results/css/css-shapes?label=master&label=stable&aligned&q=all(status%3Apass)): 4951 / 5829 (84.9%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-shapes-1): 16, [in spec.](https://drafts.csswg.org/css-shapes-1/#issues-index): 0

Notes: High interoperability, though spec. only in WD and many open issues. So it might also be put to Rough Interoperability, instead.

-----

Note that this issue only covers specs that should move to the Official Definition section. I'll go through all the other specs and create separate issues for the others sections if needed.

Sebastian

**Comment by SebastianZ** [comment]
I've added [Geometry 1](https://www.w3.org/TR/geometry-1/) to the list. Not sure about its state, though. It's in CR and interop is very high, though it it seems to have many unresolved issues. So maybe it should rather move to Rough Interoperability?

Sebastian

**Comment by dbaron** [comment]
The [Interop 2023 dashboard](https://wpt.fyi/interop-2023) might be a good list of things that have likely become broadly interoperable in the past year.

**Comment by SebastianZ** [comment]
> The [Interop 2023 dashboard](https://wpt.fyi/interop-2023) might be a good list of things that have likely become broadly interoperable in the past year.

That's true, though it mostly only covers _parts_ of specifications. I've been digging through the different items there and the tests behind them and it looks like these specs _might_ move to the Official Definition, though they still have some open issues and/or need to be published as CR(D)s:

# [CSS Containment 2](https://www.w3.org/TR/css-contain-2/)

Current status: WD (last published: 17 September 2022)
[WPT](https://wpt.fyi/results/css/css-contain?label=master&label=experimental&aligned&q=none(path%3A%2Fcss%2Fcss-contain%2Fcontainer-queries%2F)): passing tests (experimental; excluding container queries, not sure how to restrict to specific spec. level): Chrome: 842, Edge: 819, Firefox: 788, Safari: 791; [interop](https://wpt.fyi/results/css/css-contain?label=master&label=experimental&aligned&q=none(path%3A%2Fcss%2Fcss-contain%2Fcontainer-queries%2F)%20and%20all(status%3Apass)): 703 / 858 (81.9%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-contain-2): 32, [in spec.](https://drafts.csswg.org/css-contain-2/#issues-index): 0

# [Motion Path 1](https://www.w3.org/TR/motion-1/)

Current status: WD (last published: 18 December 2018)
[WPT](https://wpt.fyi/results/css/motion?label=master&label=experimental&aligned): passing tests (experimental): Chrome: 3295, Edge: 3295, Firefox: 3298, Safari: 3128; [interop](https://wpt.fyi/results/css/motion?label=master&label=experimental&aligned&q=all(status%3Apass)): 3127 / 3299 (94.8%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/fxtf-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Amotion-1): 10, [in spec.](https://drafts.fxtf.org/motion-1/#issues-index): 2

# [CSS Overscroll Behavior 1](https://www.w3.org/TR/css-overscroll-1/)

Current status: WD (last published: 6 June 2019)
[WPT](https://wpt.fyi/results/css/css-overscroll-behavior?label=master&label=stable&aligned): passing tests: Chrome: 60, Edge: 60, Firefox: 60, Safari: 60; [interop](https://wpt.fyi/results/css/css-overscroll-behavior?label=master&label=stable&aligned&q=all(status%3Apass)): 60 / 61 (98.4%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-overscroll-1): 10, [in spec.](https://drafts.csswg.org/css-overscroll-1/#issues-index): 0

# [CSS Cascade 5](https://www.w3.org/TR/css-cascade-5/)

Current status: WD (last published: 13 January 2022)
[WPT](https://wpt.fyi/results/css/css-cascade?label=master&label=experimental&aligned&q=label%3Ainterop-2022-cascade): passing tests (experimental): Chrome: 599, Edge: 599, Firefox: 617, Safari: 611; [interop](https://wpt.fyi/results/css/css-cascade?label=master&label=experimental&aligned&q=label%3Ainterop-2022-cascade%20and%20all(status%3Apass)): 558 / 620 (90.0%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-cascade-5): 10, [in spec.](https://drafts.csswg.org/css-cascade-5/#issues-index): 2

# [CSS Position 3](https://www.w3.org/TR/css-position-3/)

Current status: WD (last published: 3 April 2023)
[WPT](https://wpt.fyi/results/css/css-position?label=master&label=stable&aligned&q=none(overlay)): passing tests (excluding `overlay`, not sure how to restrict to specific spec. level): Chrome: 1134, Edge: 1100, Firefox: 1088, Safari: 1067; [interop](https://wpt.fyi/results/css/css-position?label=master&label=stable&aligned&q=none(overlay)%20and%20all(status%3Apass)): 1038 / 1144 (90.7%)
Current Snapshot section: Rough Interoperability
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-position-3): 33, [in spec.](https://drafts.csswg.org/css-position-3/#issues-index): 2

(Other specs covered in Interop 2022 and 2023 may move to the Rough Interoperability section and some features may be added to Safe to Release pre-CR Exceptions. So again, I'll file separate issues for them.)

Sebastian

**Comment by svgeesus** [comment]
I agree with all the suggested Official Definition suggestions from [your first comment](https://github.com/w3c/csswg-drafts/issues/9770#issue-2068181615) _except_ Geometry 1, which I agree would be better in Rough Interop.

**Comment by svgeesus** [comment]
For CSS Motion 1 it [needs an updated WD](https://github.com/w3c/csswg-drafts/issues/9165) which we resolved to do in August 2023. It could then go in Rough Interop.

**Comment by SebastianZ** [comment]
There's one more spec. which might be promoted:

## [CSS Shapes 1](https://www.w3.org/TR/css-shapes-1/)

Current status: WD (last published: 15 November 2022)
[WPT](https://wpt.fyi/results/css/css-shapes?label=master&label=stable&aligned): passing tests: Chrome: 4955, Edge: 4942, Firefox: 5757, Safari: 5014; [interop](https://wpt.fyi/results/css/css-shapes?label=master&label=stable&aligned&q=all(status%3Apass)): 4838 / 5773 (83.8%)
Current Snapshot section: -
Open issues: [on GitHub](https://github.com/w3c/csswg-drafts/issues?q=is%3Aopen+is%3Aissue+label%3Acss-shapes-1): 13, [in spec.](https://drafts.csswg.org/css-shapes-1/#issues-index): 0

**Comment by SebastianZ** [comment]
I've updated the statistics for all the specs listed and moved CSS Shapes 1 up to the initial comment.

As time is passing quickly, let's add this to the agenda.

I think definite candidates for the official definition are:

* CSS Scrollbars 1
* CSS Will Change 1
* Color Adjust 1
* CSS Conditional 4 (requires incorporation of CSS Conditional 3 contents; supersedes level 3 in Official Definition)
* CSS Masking 1
* CSS Overscroll Behavior 1 (requires moving to CRD)
* CSS Cascade 5 (requires moving to CRD; supersedes level 4 in Official Definition

The others listed above need more debate. (See the notes I've added.)

Sebastian

**Comment by svgeesus** [comment]
Related: 

 - https://github.com/w3c/csswg-drafts/issues/9165

**Comment by svgeesus** [comment]
Related to CSS Scrollbars 1 and degree of interop:

 - https://github.com/web-platform-tests/wpt/issues/44196
- https://github.com/web-platform-tests/wpt/issues/44185

**Comment by fantasai** [comment]
I think it's worth pointing out that the "Official Definition" section is the "This is basically a REC, just ironing out interop on a few details", i.e. the spec is extremely stable _and_ implementations are almost completely interoperable.

**Comment by fantasai** [comment]
New categorization (adding category B):
- A) REC-level stability -- might be REC, might be CR without an implementation report, or a few bugs in implementations etc.
- B) reliable CR-level stability -- should be in CR, or close _and_ has usable shipping implementations
- C) stable spec with limited implementation experience (could be destabilized once we try to implement)
- D) unreliable spec with shipping, mostly usable implementations

**Comment by svgeesus** [comment]
Closing as the [things we resolved to add](https://github.com/w3c/csswg-drafts/issues/9770#issuecomment-2474321674) have been added.
